topo_processor/util/gzip.py

```diff
@@ -0,0 +1,28 @@
+import gzip
+import os
+
+from linz_logger.logger import get_log
+
+
+def is_gzip_file(file_path: str) -> bool:
+    with open(file_path, "rb") as file:
+        # gzip magic number == "1f 8b"
+        return file.read(2) == b"\x1f\x8b"
+        # FIXME: check Header
```

Comment:
Maybe the try/except is enough? If it doesn't gzip, it's not gzipped I guess :shrug:

Reply:
I think this is fine for checking files on disk to see if they are `gzip`

Reply:
@amfage That would be okay if the `gzip` lib would return a specific `exception` when it's not a `gzipped` file, but it does not. Dealing with a generic `BadGzipFile` with a specific message for each errors (i.e. `BadGzipFile('Not a gzipped file (%r)' % magic)`) is not ideal as our code would have to parse the `error.message` to determine what if the issue. I don't like this approach because it has to rely on a `string` which potentially can change with a next release so you have to maintain it.